### ISSUE TITLE
Support for PCNTL Extension

### ISSUE THREAD

**Issue by aleej** [cloudfoundry/php-buildpack#82]
We are currently working to get [phpipam](https://github.com/phpipam/phpipam.git) deployed on Cloud Foundry.  One of the extension it needs is "pcnt" for multi-threading.  Is there a way this can be supported by the buildpack?

**Comment by hairmare** [comment]
I just looked into supporting ``pcntl`` per default...

1. #78 is requesting it as feature
2. we, as PHP devs, need to be able to require ``ext-pcntl`` and have it work with the build back
3. I have some use-cases where I see myself using the extension, not because it's the cf way to do things, but because I have some legacy software that is using ``ext-pcntl``.

I looked into hacking this into https://github.com/dmikusa-pivotal/cf-php-buildpack-binary-build-scripts but I wasn't sure if the builbpack team is still using it. I have some incoming PRs if the repo is still the way to go. Let me know.

Basically ``pcntl`` is part of php (ie. we need to install it from php source and not through pecl). We as PHP devs, want to activate it by requiring ``ext-pcntl`` in composer.json.

Personally I feel it needs supporting.

**Comment by dmikusa-pivotal** [comment]
>I looked into hacking this into https://github.com/dmikusa-pivotal/cf-php-buildpack-binary-build-scripts but I wasn't sure if the builbpack team is still using it. I have some incoming PRs if the repo is still the way to go. Let me know.

Thanks!  @flavorjones or @jtarchie should be able to confirm.  I believe they're using it or a fork of it, but I can't say that with 100% confidence.  

>Basically pcntl is part of php (ie. we need to install it from php source and not through pecl). We as PHP devs, want to activate it by requiring ext-pcntl in composer.json.

It shouldn't be hard to enable this workflow, just make sure your PR compiles it as a shared library extension.  The build pack prefers this as it keeps the size of the core php binary as small as possible.

>Personally I feel it needs supporting.

+1 don't see any reason not to support it.

**Comment by jtarchie** [comment]
@hairmare, the build scripts for all the buildpack binaries can be found in the [binary-builder](https://github.com/pivotal-cf/binary-builder) repository. These scripts are based on @dmikusa-pivotal's work.

Please let us know if you have any questions.
